lib/elastic/http.ex

```diff
@@ -83,8 +83,12 @@ defmodule Elastic.HTTP do
   def bulk(options) do
     headers = Keyword.get(options, :headers, [])
     body = Keyword.get(options, :body, "") <> "\n"
-    options = Keyword.put(options, :body, body)
     url = URI.merge(base_url(), "_bulk")
+    headers = build_auth_header(:post, url, headers, body)
```

Comment:
nit. the variables `headers` and `options` get recycles quite a bit in here. I think some name clarification could help. `auth_headers`, etc.